pandora-server-directory/src/networking/connection_client.ts

```diff
@@ -63,12 +72,45 @@ export class ClientConnection extends IncomingConnection<IDirectoryClient, IClie
 			this._account = null;
 		}
 		if (account) {
+			Assert(token?.reason === AccountTokenReason.LOGIN);
 			account.touch();
 			this._account = account;
+			this._loginTokenEventUnsubscribe?.();
```

Comment:
Shouldn't the unsubscribe happen in the higher block about removing the account - i.e. if the call is `setAccount(null)` as well?